src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Cache.cs

```diff
@@ -21,35 +22,35 @@ public void MultipleThreadsLooping()
             }
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
```

Comment:
Same comment here with the WaitAll.  It seems like it's still a useful test, as even if the code won't run in parallel, it could still run interleaved.